src/Worker/TemporalWorker.php

```diff
@@ -0,0 +1,55 @@
+<?php
+declare(strict_types=1);
+
+namespace Baldinof\RoadRunnerBundle\Worker;
+
+use Baldinof\RoadRunnerBundle\Event\WorkerStartEvent;
+use Baldinof\RoadRunnerBundle\Event\WorkerStopEvent;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Temporal\Worker\WorkerFactoryInterface;
+
+/**
+ * @internal
+ */
+final class TemporalWorker implements WorkerInterface
+{
+    private Dependencies $dependencies;
+    private WorkerFactoryInterface $workerFactory;
+    private iterable $workflows;
+    private iterable $activities;
+
+    public function __construct(
+        KernelInterface $kernel,
+        WorkerFactoryInterface $workerFactory,
+        iterable $workflows,
+        iterable $activities
+    )
+    {
+        $this->workerFactory = $workerFactory;
+
+        /** @var Dependencies $dependencies */
+        $dependencies = $kernel->getContainer()->get(Dependencies::class);
+        $this->dependencies = $dependencies;
+
+        $this->workflows = $workflows;
+        $this->activities = $activities;
+    }
+
+    public function start(): void
+    {
+        $this->dependencies->getEventDispatcher()->dispatch(new WorkerStartEvent());
+        $worker = $this->workerFactory->newWorker();
+
+        foreach ($this->workflows as $workflow) {
+            $worker->registerWorkflowTypes(get_class($workflow));
+        }
+
+        foreach ($this->activities as $activity) {
+            $worker->registerActivityImplementations($activity);
+        }
+
+        $this->workerFactory->run();
```

Comment:
It seems `run()` can accept a `HostConnectionInterface` which is created from RoadRunner, and could be retrieved in the Symfony DIC.

https://github.com/temporalio/sdk-php/blob/master/src/WorkerFactory.php#L254-L256

Reply:
[Interface](https://github.com/temporalio/sdk-php/blob/master/src/Worker/WorkerFactoryInterface.php#L47) doesn't have this parameter. It can't be used.